tests/test_core/conftest.py

```diff
@@ -75,15 +16,66 @@ async def p9(request, tmpdir):
     await ctx.close()
 
 
+@pytest.fixture
+def nsname():
+    ns = uifname()
+    netns.create(ns)
+    with IPRoute(netns=ns) as ipr:
+        ipr.link('set', index=1, state='up')
+        ipr.poll(ipr.addr, 'dump', address='127.0.0.1', timeout=5)
+    yield ns
+    try:
+        netns.delete(ns)
+    except Exception:
+        pass
+
+
+@pytest.fixture
+def link(nsname):
+    ifname = uifname()
+    with IPRoute(netns=nsname) as ipr:
+        ipr.link('add', ifname=ifname, kind='dummy', state='up')
+        (link,) = ipr.poll(ipr.link, 'dump', ifname=ifname, timeout=5)
+        yield link
+        try:
+            ipr.link('del', index=link.get('index'))
+        except NetlinkError as e:
+            if e.code != errno.ENODEV:
+                raise
+
+
+@pytest.fixture
+def index(link):
+    yield link.get('index')
+
+
+@pytest.fixture
+def ifname(link):
```

Comment:
This fixture's name is too close to the `uifname` function, which is also used in tests, which might be confusing